File: client/src/main/java/org/javadv/client/SimpleClient.java
package org.javadv.client;

import io.grpc.Channel;
import org.grpc.service.Service;
import org.grpc.service.SimpleServiceGrpc;

public class SimpleClient {
    private static final String PING = "Ping";
    private final SimpleServiceGrpc.SimpleServiceBlockingStub blockingStub;

    public SimpleClient(Channel channel) {
        blockingStub = SimpleServiceGrpc.newBlockingStub(channel);
    }

    public void sendRequest() {
        Service.SimpleRequest request = Service.SimpleRequest.newBuilder()
                .setPing(PING)
                .build();

        Service.SimpleResponse response = blockingStub.process(request);

        System.out.printf(response.getGreeting());
        System.out.println(response.getPong());
    }
}

File: kotlin-client/src/main/java/org/javadv/client/SimpleClient.kt
package org.javadv.client

import io.grpc.ManagedChannel
import org.grpc.service.Service
import org.grpc.service.SimpleServiceGrpc

class SimpleClient(channel: ManagedChannel) {
    private val blockingStub: SimpleServiceGrpc.SimpleServiceBlockingStub = SimpleServiceGrpc.newBlockingStub(channel)

    fun sendRequest() {
        val request = Service.SimpleRequest.newBuilder()
                .setPing(PING)
                .build()

        val response = blockingStub.process(request)

        println(response.greeting)
        println(response.pong)
    }

    companion object {
        private const val PING = "Ping"
    }
}